tensormap-client/src/components/DragAndDropCanvas/CustomNodes/FlattenNode/FlattenNode.js

```diff
@@ -0,0 +1,43 @@
+import {Handle,Position,useStoreApi,useReactFlow} from 'reactflow'
+
+import './FlattenNode.css'
+
+function FlattenNode({data,id }) {
+    const {setNodes} = useReactFlow();
+    const store = useStoreApi();
+    const onChange = (evt) => {
+      const { nodeInternals } = store.getState();
+      const { name, value } = evt.target;
+      setNodes(
+        Array.from(nodeInternals.values()).map((node) => {
+          if (node.id === id) {
+            node.data = {
+              ...node.data,
+              params: {
+                ...node.data.params,
+                [name]: Number(value),
+              },
+            };
+          }
+          return node;
+        })
```

Comment:
What if this `if` statement is false? are we handling that particular case?

Reply:
Hi @samyakshah3008,
Thanks a lot for your feedback. I will work on your suggestions. In here isn't the node state supposed to be updated only in the selected node? So I think there's no need to handle the `else` case. If the node id is not matching that node's state should not be changed. Please let me know if I'm wrong.

Reply:
Pleasure! @Wenuka19  ! ✨ 
Ah I see, so can we think of writing `&&` operator to execute the same functionality? What's your thought?

Reply:
Yes. But for the clarity of the code and logic I thought of going with a `if` statement.

Reply:
Ah, good point Wenuka. 
Cool, thank you for addressing my query! 😄
You can resolve this conversation. Looks good to me ✅